### ISSUE TITLE
Index was outside the bounds of the array

### ISSUE THREAD

**Issue by AdeelMalhi** [Starcounter/Home#459]
Starcounter version: `<2.3.1.7294>`.

### Issue type

- [x] Bug
- [x] Question
- [x] Urgent

### Issue description
https://starcounter.slack.com/archives/G5HAK773N/p1541589321003300

These looks like similar issues but can be different table scan not sure
https://github.com/Starcounter/Home/issues/270
https://github.com/Starcounter/Home/issues/303

IEnumerable<UrlCleanUpDetector> detectors = Db.SQL<UrlCleanUpDetector>("SELECT D FROM AdFenix.Database.UrlCleanUpDetector D WHERE Host=?", "myhost").AsEnumerable();

### Reproduction code snippet

```cs

```

### Exception stack trace

```
at Starcounter.Query.Execution.CodeGenFilterCacheShared.CreateFilter(CodeGenFilterInstrArray instrArray, DbTypeCode[] varTypes, UInt32 maxStackSize, UInt32 dataCount, RowTypeBinding typeBinding, Int32 extentNumber) in C:\TeamCity\BuildAgent\work\sc-8734\Level1\src\Starcounter\Query\Execution\CodeGeneration\CodeGenFilterCache.cs:line 202
 at Starcounter.Query.Execution.CodeGenFilterPrivate.GetFilterHandle() in C:\TeamCity\BuildAgent\work\sc-8734\Level1\src\Starcounter\Query\Execution\CodeGeneration\CodeGenFilterPrivate.cs:line 215
 at Starcounter.Query.Execution.FullTableScan.CreateEnumeratorWithNativeFilter() in C:\TeamCity\BuildAgent\work\sc-8734\Level1\src\Starcounter\Query\Execution\Enumerators\FullTableScan.cs:line 293
 at Starcounter.Query.Execution.FullTableScan.MoveNext() in C:\TeamCity\BuildAgent\work\sc-8734\Level1\src\Starcounter\Query\Execution\Enumerators\FullTableScan.cs:line 411
 at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
 at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
```

**Comment by nomanAdfenix** [comment]
:(

**Comment by bigwad** [comment]
It's a duplicate of https://github.com/Starcounter/Home/issues/270. The issue has been fixed for more than a year and the fix is available in the latest 2.3.1 release build: https://downloads.starcounter.com/download/Starcounter/LTS/2.3.1.8415

**Comment by miyconst** [comment]
Archiving.
